### DISCUSSION TITLE
Welcome to DIE-engine Discussions!

### DISCUSSION THREAD

**Discussion by horsicq** [horsicq/DIE-engine#24] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Akmb2309** [comment]
good day 

Akmb here 

i was just browsing around and wondering what exactly is  DIE_ENGINE ?

what is the purpose of it and what can i expect

**Reply by horsicq** [reply]
Hello! DiE Engine it is source codes for program Detect It Easy(DiE) You could try the program here: https://github.com/horsicq/DIE-engine/release

The description of DiE is here: https://github.com/horsicq/Detect-It-Easy/

**Reply by Akmb2309** [reply]
thanks for the quick feedback much appreciated will check it out

**Comment by DevX-Cipher** [comment]
What do you think about the idea of creating a "Die Server" — a lightweight antivirus-like service that functions similarly to Windows Defender? It would scan for malicious behavior when users download or interact with suspicious files, aiming to catch threats in real-time. @BJNFNE @horsicq.  I did create something simple here https://github.com/DevX-Cipher/Die-Features/tree/main/DieService

**Reply by BJNFNE** [reply]
I like the Idea, maybe we can even implement this into the Telegram bot.

**Reply by BJNFNE** [reply]
Invalid link :(

**Reply by DevX-Cipher** [reply]
@BJNFNE i fixed the link sorry 

the Telegram bot. However, one thing to consider is that users would need to have Telegram installed and an active Telegram account to use the service, which might limit accessibility for some—especially in environments where installing apps or creating accounts is restricted. @DosX-dev

**Comment by DevX-Cipher** [comment]
Add a Windows Shell Property Sheet Extension for DIE
What do you think about integrating DIE into the Windows Properties dialog? The idea is to provide direct metadata access from Windows Explorer, so when users right-click on a suspicious file and select "Properties", a new tab would appear showing DIE’s analysis — such as packer information, signatures, entropy, and other metadata — without needing to open the full DIE application.

This could streamline workflows for malware analysts and developers by offering quick insights directly in the shell.
@BJNFNE @horsicq  example here https://github.com/DevX-Cipher/Die-Features/tree/main/DieProperty

**Comment by DevX-Cipher** [comment]
ntroduce a Plugin System for DIE
One possible direction to consider is implementing a plugin architecture for DIE. This could significantly reduce the development burden on the core maintainer(s) by allowing the community to contribute additional functionality through plugins.

Community-created plugins that prove to be stable and useful could later be considered for integration into the core application, following proper review. This would allow DIE to evolve more flexibly, support a wider range of formats and analysis techniques, and stay relevant to various use cases — all without bloating the main codebase.

It could also foster a stronger open-source ecosystem around DIE.

Example here not the best https://github.com/DevX-Cipher/Die-Features/tree/main/Plugins
Would this be something worth exploring?
@BJNFNE @horsicq

**Reply by DosX-dev** [reply]
DiE had a plugin system but it was removed due to being unnecessary

**Reply by DevX-Cipher** [reply]
@DosX-dev  never knew good to know
